client/src/components/AttributeTrackFilters.vue

```diff
@@ -0,0 +1,466 @@
+import {
+  computed, defineComponent, ref, Ref,
+} from '@vue/composition-api';
+import { AttributeTrackFilter, MatchOperator } from 'vue-media-annotator/AttributeTrackFilterControls';
+import AttributeTrackFilterVue from '../../dive-common/components/AttributeTrackFilter.vue';
+import { useAttributes, useTrackFilters, useTrackStyleManager } from '../provides';
+
+
+export default defineComponent({
+  name: 'AttributeTrackFilters',
+
+  props: {
+  },
+
+  components: {
+    AttributeTrackFilterVue,
+  },
+
+  setup() {
+    const trackFilters = useTrackFilters();
+    const typeStylingRef = useTrackStyleManager().typeStyling;
+    const types = computed(() => trackFilters.allTypes.value);
+    const attributes = useAttributes();
+    const attributeTypes = computed(() => {
+      const typeMap: Record<string, string> = {};
+      attributes.value.forEach((item) => {
+        typeMap[item.name] = item.datatype;
+      });
+      return typeMap;
+    });
+
+
+    const editingFilter: Ref<null | number> = ref(null);
+    const addFilterDialog = ref(false);
+    const filters = trackFilters.attributeFilters;
+
+    const existingNames = computed(() => trackFilters.attributeFilters.value.map((item) => item.name));
+    // editing/adding Attribute Filter
+    const editingAtrKey = ref('');
+    const editName = ref('');
+    const baseOps = ['=', '!=', '>', '<', '>=', '<=', 'range', 'in', 'rangeFilter'];
+    const editingAttributeType = computed(() => {
+      if (editingAtrKey.value) {
+        const filtered = attributes.value.filter((item) => {
+          if (editingAtrKey.value) {
+            return item.name === editingAtrKey.value;
+          }
+          return false;
+        });
+        if (filtered.length > 0) {
+          return filtered[0].datatype;
+        }
+      }
+      return null;
+    });
+
+    const editingOps = computed(() => {
+      if (editingAttributeType.value) {
+        if (editingAttributeType.value === 'number') {
+          return ['=', '!=', '>', '<', '>=', '<=', 'range', 'rangeFilter'];
+        } if (editingAttributeType.value === 'text') {
+          return ['=', '!=', 'in', 'contains'];
+        }
+        if (editingAttributeType.value === 'boolean') {
+          return ['=', '!='];
+        }
+      }
+      return baseOps;
+    });
+
+
+    const editingAtrOp: Ref<MatchOperator> = ref('=');
+    const editingAtrVal: Ref<string[] | string | number | number[] | null | boolean> = ref('');
+    const editingTypeFilter: Ref<string[]> = ref([]);
+    const editingRange: Ref<number[]> = ref([0, 1]);
+    const editingAtrTypeList = ref(['track', 'detection']);
+    const editingFilterEnabled = ref(true);
+    const editingAtrType: Ref<'track' | 'detection'> = ref('track');
+    const editingUserDefined = ref(true);
+    const editingPrimaryDisplay = ref(true);
+    const attributeList = computed(() => attributes.value.filter((item) => item.belongs === editingAtrType.value).map((item) => item.name));
```

Comment:
I think it would be more intuitive if these values were all in one dict ie.
```
const editingFilter = ref({
atrKey: '',
name: '',
... }

Reply:
To maintain reactivity I think It would have to be wrapped `reactive()` with a type declaration above it.  Something like what I have below.
```
    interface EditingOptions {
      atrKey: string;
      name: string;
      atrOp: MatchOperator;
      atrVal: string[] | string | number | number[] | null | boolean;
      typeFilter: string[];
      range: number[];
      atrType: 'track' | 'detection';
      filterEnabled: boolean;
      userDefined: boolean;
      primaryDisplay: boolean;
    }
    const editing: EditingOptions = reactive({
      atrKey: '',
      name: '',
      atrOp: '=',
      atrVal: '',
      typeFilter: [],
      range: [0,1],
      atrType: 'track',
      filterEnabled: true,
      userDefined: true,
      primaryDisplay: true,
    })
    const dropdowns = reactive({
      atrTypeList: ['track', 'detection'],
      ops: ['=', '!=', '>', '<', '>=', '<=', 'range', 'in', 'rangeFilter'],
    })
```
And sense Github doesn't do highlighting in code snippets:
![image](https://github.com/Kitware/dive/assets/61746913/b4a2e251-ca28-45f0-a410-512dd4051c3c)


    I'd separate out the static values for the dropdowns into their own reactive.  Does this look better?

Reply:
I don't think it needs to be reactive. I think its actually preferred to use ref even in this case:

https://vuejs.org/guide/essentials/reactivity-fundamentals.html#limitations-of-reactive

I do agree that there needs to be the type definition and yes, I think this is better.

Reply:
You are right, for some reason I didn't think the reactivity was mapped depp:
![image](https://github.com/Kitware/dive/assets/61746913/31c3e8ab-6aa1-4384-9b4d-c95227e15c0d)